#167891136 Modify validations and implement feedback

amkayondo opened the pull request:
#### What does this PR do?
Modifies validations and implement feedback
#### Description of Task to be completed?
- validate data types on user input
       -  post trips
       - get trips by origin and destination
       - get trip by id
- add min and max to seat capacity
- fix bug of available seats
- user should not see cancelled trips
#### How should this be manually tested?
send invalid details to any field of the app
#### Any background context you want to provide?
`400` Bad Request
#### What are the relevant pivotal tracker stories?
[#167891136](https://www.pivotaltracker.com/story/show/167891136)
#### Screenshots
![Screenshot (59)](https://user-images.githubusercontent.com/46356108/63019304-3468c300-be9b-11e9-801f-9716c6e83fde.png)
![Screenshot (60)](https://user-images.githubusercontent.com/46356108/63019305-3468c300-be9b-11e9-94ef-1bdc23cb15e1.png)
![Screenshot (61)](https://user-images.githubusercontent.com/46356108/63019306-35015980-be9b-11e9-9ab1-504d4a6c058c.png)
![Screenshot (62)](https://user-images.githubusercontent.com/46356108/63019308-35015980-be9b-11e9-9cb6-a9088162f7a8.png)
![Screenshot (63)](https://user-images.githubusercontent.com/46356108/63019309-35015980-be9b-11e9-832c-708dc6b073b5.png)
![Screenshot (64)](https://user-images.githubusercontent.com/46356108/63019310-3599f000-be9b-11e9-9915-4e2d60693326.png)
![Screenshot (58)](https://user-images.githubusercontent.com/46356108/63019311-3599f000-be9b-11e9-842f-47224dd9570a.png)

amkayondo committed:
chore(API) Modify validations and implement feedback
- validate data types
- validate trip id
- user shouldnt see cancelled trips

Kasulejoseph commented on server/middleware/validators/validateInput.js:
```diff
@@ -3,30 +3,32 @@ import resPonse from '../../helpers/responses/response';
   trip(req, res, next) {
     const result = validate(req.body, {
-      seating_capacity: Joi.string().trim().regex(regeX)
+      seating_capacity: Joi.number().integer().min(50).max(80)
```
50 is too high, consider having taxis as well as your implement this

Kasulejoseph commented on server/middleware/validators/validateInput.js:
```diff
@@ -3,30 +3,32 @@ import resPonse from '../../helpers/responses/response';
         .required()
-        .label('Only Integers are allowed'),
-      bus_license_number: Joi.string().trim().alphanum()
+        .label('Seating capacity must be Only Integers with between 50 - 80'),
```
50 is to high

Kasulejoseph commented on server/middleware/validators/validateInput.js:
```diff
@@ -3,30 +3,32 @@ import resPonse from '../../helpers/responses/response';
+        .max(50000)
         .required()
-        .label('Fare should be an integer'),
+        .label('Fare should be an integer and should be betwwen 15000 - 50000'),
```
Check the typo on `betwwen`.

amkayondo committed:
chore(API): implemenet feedback
- decreased the min seating_capacity to 14
- admin cant cancel a cancelled trip
[starts #167891136] [finishes #167834077, #167865032]

amkayondo committed:
chore(API): implement feedback
- cahange betwwen to between

amkayondo committed:
chore(API) Fix bugs
- fix th null value on availableseats
- admin cant cancel a trip twice
- users cant see cancelled trips

Kasulejoseph reviewed (approved):
Looks good.
